internal/modules/nativeext/nativeext.go

```diff
@@ -160,20 +185,16 @@ func (h *Host) RegisterInprocExtension(name string, ext apiv1.Implementation) {
 
 // getExtensionPath returns the path to an extension binary
 func (h *Host) getExtensionPath(version *resolver.Version, name string) (string, error) {
-	cacheDir := os.Getenv("XDG_CACHE_HOME")
-	if cacheDir == "" { // default to $HOME/.cache as per XDG spec
-		homeDir, err := os.UserHomeDir()
-		if err != nil {
-			return "", fmt.Errorf("failed to get home directory: %w", err)
-		}
-		cacheDir = filepath.Join(homeDir, ".cache")
+	cacheDir, err := getCacheDir()
+	if err != nil {
+		return "", err
 	}
 
 	// Example:
 	//
 	// $XDG_CACHE_HOME/stencil/nativeexts/github.com--rgst-io--plugin/v1.3.0/plugin
 	path := filepath.Join(
-		cacheDir, "stencil", "nativeexts",
```

Comment:
isn't it still better to keep this layer in here since it lasts a while?

Reply:
wdym? I only moved where "stencil" got added since i pulled out the cache dir logic into its own function

Reply:
oh balls, i didn't see the stealth filepath join at the bottom of the getcachedir func